src/ProtectedRoomsSet.ts

```diff
@@ -241,6 +240,21 @@ export class ProtectedRoomsSet {
         }
     }
 
+    /**
+     * Sync all the rooms with all the watched lists, banning and applying any changed ACLS.
+     */
+    public async syncLists() {
+        for (const list of this.policyLists) {
+            const { revisionId } = await list.updateList();
+            const previousRevision = this.listRevisions.get(list);
+            if (previousRevision === undefined || revisionId > previousRevision) {
```

Comment:
Well, I've just looked up the definition of ULID and I guess this comparison works, but:

1. Since it's very implicit, it's also very fragile. I'd be much more comfortable if we had a dedicated data structure with a dedicated comparison method.
2. That would need to be very documented.